File: app/src/main/java/com/aresid/happyapp/signup/SubscribeFragment.java
package com.aresid.happyapp.signup;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.aresid.happyapp.R;

/**
 * Created on: 15/04/2020
 * For Project: HappyApp
 * Author: René Spies
 * Copyright: © 2020 ARES ID
 */
public class SubscribeFragment
		extends Fragment {
	
	private static final String TAG = "SubscribeFragment";
	
	public SubscribeFragment() {
		
		Log.d(TAG, "SubscribeFragment: called");
		// Required public empty constructor
		
	}
	
	@Override
	public void onCreate(@Nullable Bundle savedInstanceState) {
		
		Log.d(TAG, "onCreate: called");
		
		super.onCreate(savedInstanceState);
		
	}
	
	@Nullable
	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
			@Nullable Bundle savedInstanceState) {
		
		Log.d(TAG, "onCreateView: called");
		
		// Inflate layout
		return inflater.inflate(R.layout.fragment_subscribe, container, false);
		
	}
	
}

File: app/src/main/java/com/aresid/happyapp/signup/SubscribeFragment.kt
package com.aresid.happyapp.signup

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.aresid.happyapp.databinding.FragmentSubscribeBinding
import timber.log.Timber

/**
 * Created on: 15/04/2020
 * For Project: HappyApp
 * Author: René Spies
 * Copyright: © 2020 ARES ID
 */
class SubscribeFragment: Fragment() {
	
	// Binding for the layout
	private lateinit var binding: FragmentSubscribeBinding
	
	// ViewModel
	private lateinit var subscribeFragmentViewModel: SubscribeViewModel
	
	override fun onCreateView(
		inflater: LayoutInflater,
		container: ViewGroup?,
		savedInstanceState: Bundle?
	): View? {
		
		Timber.d("onCreateView: called")
		
		// Define the binding
		binding = FragmentSubscribeBinding.inflate(
			inflater,
			container,
			false
		)
		
		// Define the ViewModel
		subscribeFragmentViewModel = ViewModelProvider(this).get(SubscribeViewModel::class.java)
		
		// Return the inflated layout
		return binding.root
		
	}
	
}